jupyter_book/book_template/content/features/notebooks.ipynb

```diff
@@ -491,11 +491,69 @@
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
+   "cell_type": "markdown",
    "metadata": {},
-   "outputs": [],
-   "source": []
+   "source": [
+    "## Executing notebooks\n",
+    "\n",
+    "✨**experimental**✨\n",
+    "\n",
+    "By default, Jupyter Book will not run any of your notebooks when you build your\n",
+    "book. It assumes you have already run each one, and simply converts each page to\n",
+    "HTML.\n",
+    "\n",
+    "However, sometimes you'd like to store *just* the book content, and then *run* this content\n",
+    "when you build your book. This helps you ensure that your content still runs, and\n",
+    "that you've always got the latest outputs in your book. The drawback to this is that\n",
+    "your book will take longer to build, especially if your content takes a long time\n",
+    "to run."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Note:** If you store your Notebook content as [Jupytext documents](jupytext.html), then\n",
+    "they will automatically be run each time you build your book. This is because Jupytext\n",
+    "documents don't contain any outputs at all, and we make the assumption that they should\n",
+    "be run first before finishing your book."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It's possible to **execute your book content at build time**. There are a few convenience\n",
+    "functions and options for this:\n",
+    "\n",
+    "### **`jupyter-book run`**\n",
+    "\n",
+    "This is a command-line function that allows you to quickly execute pages of\n",
+    "your book and store the content in-line. You can use it with a path to a single\n",
+    "book page like so:\n",
+    "\n",
+    "```\n",
+    "jupyter-book run path/to/notebook.ipynb\n",
+    "```\n",
+    "\n",
+    "And it will simply run the execute in-place.\n",
+    "\n",
+    "You can also specify a folder with a **collection** of pages as the first argument:\n",
+    "\n",
+    "```\n",
+    "jupyter-book run path/to/notebook_folder\n",
+    "```\n",
+    "\n",
+    "In this case, all notebooks in this folder and sub-folders will be run in-place.\n",
+    "\n",
+    "### **`jupyter-book build --execute`**\n",
+    "\n",
+    "Another option is to pass the `--execute` flag when you build your book. This\n",
+    "will cause each page to be **executed** when it is built. Remember that the HTML\n",
+    "for pages is cached until you update your content, so while this will take a long\n",
```

Comment:
> Remember that the HTML for pages is cached until you update your content, so while this will take a long time the first time you run it (since all your book content will need to run), it\ should be incrementally faster in subsequent builds.

To me this suggests that `--execute` runs in a cached way, so it would need to keep track of each `ipynb` and only run it again if it had changed?